tests/fixtures/modifiers.js

```diff
@@ -263,6 +263,29 @@ const modifiersFixtures = [
 		'expected': '([^](?:.))',
 		'expectedFlags': ''
 	},
+	{
+		'pattern': '(.(?-s:.))',
+		'flags': 's',
+		'options': { dotAllFlag: 'transform', modifiers: 'transform' },
+		'expected': '([^](?:.))',
```

Comment:
In current main, it is transpiled as `([^](?:[^]))`, which is incorrect as the dotAll transform should not be applied when `modifiersData.s` is `false`.